include/psa/crypto_struct.h

```diff
@@ -447,12 +447,15 @@ static inline psa_algorithm_t psa_get_key_algorithm(
     return attributes->MBEDTLS_PRIVATE(core).MBEDTLS_PRIVATE(policy).MBEDTLS_PRIVATE(alg);
 }
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wredundant-decls"
```

Comment:
What happens when this is compiled with other compilers, such as clang and MSVC?

Reply:
It's either supported (https://clang.llvm.org/docs/UsersManual.html#controlling-diagnostics-via-pragmas) or issues a warning (https://learn.microsoft.com/en-us/cpp/preprocessor/pragma-directives-and-the-pragma-keyword?view=msvc-170#remarks).

Reply:
I do see the MSVC tests treat warnings as errors and fail. I'll look into preventing that.

Reply:
> What happens when this is compiled with other compilers, such as clang and MSVC?

259a17ac8fbcbde66ebabe7f94de5dfee976ad69 seems to fix this issue.